packages/quantic/force-app/main/default/lwc/quanticModal/quanticModal.js

```diff
@@ -94,4 +103,51 @@ export default class QuanticModal extends LightningElement {
   get tabindex() {
     return this.isVisible ? 0 : -1;
   }
+
+  get allAssignedElements() {
+    const headerSlot = this.template.querySelector('slot[name=header]');
+    const contentSlot = this.template.querySelector('slot[name=content]');
+    const footerSlot = this.template.querySelector('slot[name=footer]');
+
+    const allAssignedElements = [
+      // @ts-ignore
+      ...headerSlot.assignedElements(),
+      // @ts-ignore
+      ...contentSlot.assignedElements(),
+      // @ts-ignore
+      ...footerSlot.assignedElements(),
+    ].map((element) => Array.from([element, ...element.childNodes]));
+    return this.flattenArray(allAssignedElements);
+  }
+
+  focusOnLastElement() {
+    for (const element of this.allAssignedElements.reverse()) {
+      const lastFocusableElement = getLastFocusableElement(element);
+      if (lastFocusableElement) {
+        // @ts-ignore
+        lastFocusableElement.focus();
+        break;
+      }
+    }
+  }
+
+  focusOnFirstElement() {
+    for (const element of this.allAssignedElements) {
+      const firstFocusableElement = getFirstFocusableElement(element);
+      if (firstFocusableElement) {
+        // @ts-ignore
+        firstFocusableElement.focus();
+        break;
+      }
+    }
+  }
+
+  /**
+   * Flattens an array.
+   * @param {array} array 
+   * @returns array
+   */
+  flattenArray(array){
```

Comment:
This exists maybe it does what you are looking for?

[Array.flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat)

In any case, I don't think a method that flattens an array should live in the `quanticModal` component ;)

Reply:
I was using the js `flat` method at first, and then found out that it was causing some errors in our unit tests, apparently typescript didn't really like this flat method.
So I had the choice between updating the tsconfig file to make our ts accept the flat method or to just recreate this method my self and avoid the risk of breaking other stuff if I mess up the tsconfig x)